AsyncAwait.Task1.CancellationTokens/Calculator.cs

```diff
@@ -1,21 +1,25 @@
 ﻿using System.Threading;
+using System.Threading.Tasks;
 
 namespace AsyncAwait.Task1.CancellationTokens;
 
 internal static class Calculator
 {
     // todo: change this method to support cancellation token
-    public static long Calculate(int n /*, CancellationToken token*/)
-    {
-        long sum = 0;
-
-        for (var i = 0; i < n; i++)
+    public static Task<long> CalculateAsync(int n , CancellationToken token) =>
+        Task.Factory.StartNew(() =>
         {
-            // i + 1 is to allow 2147483647 (Max(Int32)) 
-            sum = sum + (i + 1);
-            Thread.Sleep(10);
-        }
+            long sum = 0;
+
+            for (var i = 0; i < n; i++)
+            {
+                token.ThrowIfCancellationRequested();
+
+                // i + 1 is to allow 2147483647 (Max(Int32)) 
+                sum += (i + 1);
+                Thread.Sleep(10);
```

Comment:
Please use async replacement for Thread.Sleep

Reply:
Changed to `await Task.Delay(10)`